### DISCUSSION TITLE
Python library support for reserveohio

### DISCUSSION THREAD

**Discussion by kmulh-1** [juftin/camply#307] [General]
I would like to use the python library in my own app to search for sites on reserveohio.com.  It looks like only Yellowstone and recreationdotgov are currently supported in the libarary, do you have any plans to add additional providers?

**Comment by juftin** [comment]
Hey @kmulh-1 - the Python library does support ReserveOhio but the Python API isn't completely documented since camply is first and foremost a CLI. 

All Python usage of Camply requires a `Search` class. See ReserveOhios's Search class here: https://juftin.com/camply/reference/search/search_usedirect/#camply.search.search_usedirect.SearchOhioStateParks

All `Search` classes can be imported from the `camply.search` namespace:

```python
from camply.search import SearchOhioStateParks
```

And see the very basic Python API documentation here: https://juftin.com/camply/python/ - all `Search`  classes share the same fundamental methods.

**Reply by juftin** [reply]
Every `Search` class also has its own `Provider` class as well. The `Provider` class provides lower level functionality like searching for recreation areas:

https://juftin.com/camply/reference/providers/#camply.providers.OhioStateParks

```python
from camply.providers import OhioStateParks
```

**Reply by juftin** [reply]
Here's some more sample Python code:

```python
from __future__ import annotations

from camply.containers import CampgroundFacility, RecreationArea
from camply.providers import OhioStateParks

provider = OhioStateParks()
recreation_areas: list[RecreationArea] = provider.search_for_recreation_areas(
    query="Park"
)
campgrounds: list[CampgroundFacility] = provider.find_campgrounds(
    rec_area_id=[recreation_areas[0].recreation_area_id]
)
```

I recommend using a modern Python IDE to explore other available methods and params 
<img width="924" alt="image" src="https://github.com/juftin/camply/assets/49741340/5da71587-b8bf-48c7-b5ca-93e9deb7272f">
